superset/assets/src/dashboard/util/charts/getFormDataWithExtraFilters.js

```diff
@@ -23,42 +23,78 @@ import getEffectiveExtraFilters from './getEffectiveExtraFilters';
 const cachedDashboardMetadataByChart = {};
 const cachedFiltersByChart = {};
 const cachedFormdataByChart = {};
-
-const getExtraFilters = (chart, globalFilters) => {
-  const slicesInState = getLinkedSliceWithFilters(chart.formData,globalFilters)
-  const filters = getFiltersFromSlices(slicesInState,globalFilters)
-  return filters;
+const filterKeys = [
+  '__time_range',
+  '__time_col',
+  '__time_grain',
+  '__time_origin',
+  '__granularity',
+];
+const getExtraFilters = (subscriberMap, globalFilters, slicesInState) => {
+    if (subscriberMap && subscriberMap.actions.indexOf("APPLY_FILTER") > -1) {
+      const filters = getFiltersFromSlices(slicesInState,globalFilters)
+      return filters;
+    }
+    return {};
 }
-
+  
 const getFiltersFromSlices = (slices, globalFilters) => {
   const filters = {};
   slices.forEach( slice => {
-    filters[slice] = globalFilters[slice];
+    const sliceId = Object.keys(slice) && Object.keys(slice)[0];
+    const subscribe_columns = slice[sliceId]
+    filters[sliceId] = {}
+    subscribe_columns.forEach( item => {
+      if (globalFilters[sliceId].hasOwnProperty(item.col)) {
+        filters[sliceId][item.col] = globalFilters[sliceId][item.col];
+      }
+    })
+    // handle publish special filterKeys column seperately
+    filterKeys.forEach(element => {
+      if (globalFilters[sliceId].hasOwnProperty(element)) {
+        filters[sliceId][element] = globalFilters[sliceId][element];
+      }
+    });
+    
   })
   return filters;
 }
 
-const getLinkedSliceWithFilters = (formData, globalFilters) => {
-  let slicesInState = [];
-  const propExist = formData.hasOwnProperty("linked_slice");
-  if(propExist) {
-    const linked_slices = formData.linked_slice;
-    if (linked_slices instanceof Array) {
-      slicesInState  = linked_slices.filter(element => globalFilters.hasOwnProperty(element));  
+const getLinkedSlicesExistInFilters = (subscriberMap, globalFilters) => {
+  let linkedSlicesExistInFilters = [];
```

Comment:
can we rename getLinkedSlicesExistInFilters to getPubSlicesOfSubscribeSlicesMatchesInFilters